vault status panics for HA server when /sys/leader returns 500

**Environment:**

* Vault Version: Vault v0.9.3 ('5acd6a21d5a69ab49d0f7c0bf540123a9b2c696d')
* Operating System/Architecture: OSX

**Expected Behavior:**
Display error from server rather than panicking.

**Actual Behavior:**
Panics.

**Steps to Reproduce:**
<!-- List the steps required to reproduce the issue. -->

1. Get your HA cluster into a state where leader election is busted (e.g. my server logs are spamming `[ERROR] core: failed to acquire lock: error=failed to create session: Unexpected response code: 500 (No known Consul servers)`)

2. Try to check the Vault status.

```
$ curl --cacert $VAULT_CACERT $VAULT_ADDR/v1/sys/leader
{"errors":["Unexpected response code: 500"]}

$ vault status
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x25c6624]

goroutine 1 [running]:
github.com/hashicorp/vault/command.OutputSealStatus(0x3d1e2a0, 0xc420097e50, 0xc4201c36c0, 0xc4205dc980, 0x0)
        /private/tmp/vault-20180128-5063-cdu8kz/src/github.com/hashicorp/vault/command/format.go:244 +0x6d4
github.com/hashicorp/vault/command.(*StatusCommand).Run(0xc4201a93e0, 0xc420098010, 0x0, 0x0, 0xc42063e2a0)
        /private/tmp/vault-20180128-5063-cdu8kz/src/github.com/hashicorp/vault/command/status.go:82 +0x1e2
github.com/hashicorp/vault/vendor/github.com/mitchellh/cli.(*CLI).Run(0xc42042db80, 0xc42042db80, 0xc420640c40, 0xc420180780)
        /private/tmp/vault-20180128-5063-cdu8kz/src/github.com/hashicorp/vault/vendor/github.com/mitchellh/cli/cli.go:255 +0x1eb
github.com/hashicorp/vault/command.Run(0xc420098010, 0x1, 0x1, 0x2622fb9)
        /private/tmp/vault-20180128-5063-cdu8kz/src/github.com/hashicorp/vault/command/main.go:47 +0x411
main.main()
        /private/tmp/vault-20180128-5063-cdu8kz/src/github.com/hashicorp/vault/main.go:10 +0x63
```

**Important Factoids:**
<!-- Describe any atypical environment setup, if any. -->

I'm pretty sure the HA setup problem is user error. This issue is just about the client panic on `vault status`

--- a/command/format.go
+++ b/command/format.go
@@ -298,6 +298,11 @@ func OutputSealStatus(ui cli.Ui, client *api.Client, status *api.SealStatusRespo
 	leaderStatus, err := client.Sys().Leader()
 	if err != nil && strings.Contains(err.Error(), "Vault is sealed") {
 		leaderStatus = &api.LeaderResponse{HAEnabled: true}
+		err = nil
+	}
+	if err != nil {
+		ui.Error(fmt.Sprintf("Error checking leader status: %s", err))
+		return 1
 	}
 
 	// Output if HA is enabled
@@ -305,16 +310,23 @@ func OutputSealStatus(ui cli.Ui, client *api.Client, status *api.SealStatusRespo
 	if leaderStatus.HAEnabled {
 		mode := "sealed"
 		if !status.Sealed {
+			out = append(out, fmt.Sprintf("HA Cluster | %s", leaderStatus.LeaderClusterAddress))
 			mode = "standby"
+			showLeaderAddr := false
 			if leaderStatus.IsSelf {
 				mode = "active"
+			} else {
+				if leaderStatus.LeaderAddress == "" {
+					leaderStatus.LeaderAddress = "<none>"
+				}
+				showLeaderAddr = true
 			}
-		}
+			out = append(out, fmt.Sprintf("HA Mode | %s", mode))
 
-		out = append(out, fmt.Sprintf("HA Mode | %s", mode))
-
-		if !status.Sealed {
-			out = append(out, fmt.Sprintf("HA Cluster | %s", leaderStatus.LeaderClusterAddress))
+			// This is down here just to keep ordering consistent
+			if showLeaderAddr {
+				out = append(out, fmt.Sprintf("Active Node Address: | %s", leaderStatus.LeaderAddress))
+			}
 		}
 	}
 

